src/main/java/org/logstash/plugins/inputs/http/MessageProcessor.java

```diff
@@ -35,11 +44,65 @@ public void onRejection() {
     @Override
     public void run() {
         try {
-            final FullHttpResponse response = messageHandler.onNewMessage(remoteAddress, req);
+            final HttpResponse response;
+            final String token = req.headers().get(HttpHeaderNames.AUTHORIZATION);
+            if (messageHandler.validatesToken(token)) {
```

Comment:
may want to short circuit this if
```
if (token != null && !token.isEmpty() && messageHandler.validatesToken(token))
```

Reply:
unfortunately we can't without moving more logic to java, because `validatesToken` has the reponsibility to both validate the token but only if it has been set in the configuration. short circuiting like this would mean that if a user doesn't set up creds in logstash any request without auth will return 401